### PR TITLE
Fixing accessibility issues with section links (DAC audit) A

### PR DESCRIPTION
**Pull request by marianayovcheva**
https://citizensadvice.atlassian.net/browse/NP-2768
https://citizensadvice.atlassian.net/browse/NP-2769

The DAC audit has flagged two accessibility issues with the section links:
- Visual headings - Visual headings were found which were not programmatically marked up for users of screen reading assistive technologies (p.13)
- Incorrect list mark-up - There was a list found that does not correctly list all the items. (p.15)

This PR is based on the recommendations in the audit.

### CODE REVIEW

File: engine/app/components/citizens_advice_components/section_links.html.haml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-      - if section_title_url.present?
-        = link_to section_title, section_title_url, "data-testid": "section-title-link"
-      - else
-        = section_title
+    %h2.cads-section-links__title= title
     %nav{ aria: { label: section_title } }
       %ul.cads-section-links__links
+        %li.cads-section-links__section-title-link
+          - if section_title_url.present?
+            = link_to section_title, section_title_url, "data-testid": "section-title-link"
```
[the remark this answers is not in the document]
Comment (marianayovcheva) [reply engine/app/components/citizens_advice_components/section_links.html.haml:8]: To clarify, do you mean having the `h3` inside the `li`?

[the remark this answers is not in the document]
Comment (davidsauntson) [reply engine/app/components/citizens_advice_components/section_links.html.haml:8]: > To clarify, do you mean having the `h3` inside the `li`?

Yup that is what I meant 👍🏻

[the remark this answers is not in the document]
Comment (marianayovcheva) [reply engine/app/components/citizens_advice_components/section_links.html.haml]: @davidsauntson I've now updated this to include h3

**Review by davidsauntson — APPROVED** [review]
🍎

**Pull request merged**